### PR TITLE
Handle sharing props better in case FF is turned off

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**
JIRA: TNT-332
When creating new dashboard via Save or Save As, don't generate sharing props when feature turned off.
When creating copy of locked dashboard lock flag needs to be always cleaned .

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5701/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7433/ : FAILURE in 7m 19s

File: examples/playground/src/App.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import bearFactory, {
     AnonymousAuthProvider,
```
Comment (ivanmjartan) [inline examples/playground/src/App.tsx:5]: remove changes in playground

File: examples/playground/webpack.config.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ const playgroundDefinitions = {
     developer: ["https://developer.na.gooddata.com", "xms7ga4tf3g3nzucd8380o2bev8oeknp"],
     public: ["https://live-examples-proxy.herokuapp.com", "xms7ga4tf3g3nzucd8380o2bev8oeknp"],
```
Comment (ivanmjartan) [inline examples/playground/webpack.config.js:22]: remove changes in playground

File: examples/playground/src/App.tsx [hunk 3]
```diff
@@ -38,7 +39,7 @@ export const App: React.FC = () => {
     return (
         <BackendProvider backend={backend}>
             <WorkspaceProvider workspace={WORKSPACE}>
-                {/* Build your playground components under the playground directory.*/}
+                <DashboardExample />
```
Comment (xMort) [inline examples/playground/src/App.tsx:42]: Should this be merged?

File: examples/playground/webpack.config.js [hunk 4]
```diff
@@ -21,6 +21,7 @@ const playgroundDefinitions = {
...
     reference: ["https://secure.gooddata.com", "l32xdyl4bjuzgf9kkqr2avl55gtuyjwf"],
+    ine: ["https://staging3.intgdc.com", "auiwj6pa2cs3twpjr98gtjfb34x3i0gv"],
```
Comment (xMort) [inline examples/playground/webpack.config.js:24]: Should this be merged?

**Review by xMort — APPROVED** [review]

**Comment by xMort** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1877

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1617/ : SUCCESS in 2m 22s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1897/ : SUCCESS in 44s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/694/ : SUCCESS in 19m 03s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/7438/ : SUCCESS in 21m 15s

**Pull request merged**